### PR TITLE
Implemented jwt token security

### PR DESCRIPTION
**Pull request by pikador10**

### CODE REVIEW

File: src/main/java/com/hw/bookstore/security/auth/jwt/TokenAuthenticationFilter.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+                var authentication = new UsernamePasswordAuthenticationToken(
+                        userDetails, null, userDetails.getAuthorities());
+                authentication.setDetails(new WebAuthenticationDetailsSource()
+                        .buildDetails(request));
+                SecurityContextHolder.getContext().setAuthentication(authentication);
+            }
+        } catch (Exception ex) {
+            log.error("Token validation error", ex);
+            throw ex;
```
Comment (liliia-ponomarenko) [inline src/main/java/com/hw/bookstore/security/auth/jwt/TokenAuthenticationFilter.java]: better throw custom exception

File: src/main/java/com/hw/bookstore/security/auth/jwt/TokenAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        }
+
+        filterChain.doFilter(request, response);
+    }
+
+    private String getJwtFromRequest(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
```
Comment (liliia-ponomarenko) [inline src/main/java/com/hw/bookstore/security/auth/jwt/TokenAuthenticationFilter.java]: You can use the HttpHeader constant

File: src/main/java/com/hw/bookstore/security/auth/jwt/TokenProvider.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                    .parseSignedClaims(token);
+            return true;
+        } catch (MalformedJwtException ex) {
+            log.error("Invalid JWT token");
+        } catch (ExpiredJwtException ex) {
+            log.error("Expired JWT token");
+        } catch (UnsupportedJwtException ex) {
+            log.error("Unsupported JWT token");
+        } catch (IllegalArgumentException ex) {
+            log.error("JWT claims string is empty.");
```
Comment (liliia-ponomarenko) [inline src/main/java/com/hw/bookstore/security/auth/jwt/TokenProvider.java]: Should you throw an invalid token exception?

**Review by liliia-ponomarenko — APPROVED** [review]
Great! A few minor comments ;)

File: src/main/java/com/hw/bookstore/security/auth/jwt/TokenAuthenticationFilter.java (hunk 1 above)
Comment (pikador10) [reply src/main/java/com/hw/bookstore/security/auth/jwt/TokenAuthenticationFilter.java]: fix

**Pull request merged**